src/org/sosy_lab/java_smt/solvers/z3/Z3ArrayFormulaManager.java

```diff
@@ -45,6 +46,22 @@ protected <TI extends Formula, TE extends Formula> Long internalMakeArray(
     return getFormulaCreator().makeVariable(z3ArrayType, pName);
   }
 
+  @Override
+  protected <TI extends Formula, TE extends Formula> Long internalMakeArray(
+      FormulaType<TI> pIndexType, FormulaType<TE> pElementType) {
+    long func =
+        getFormulaCreator()
+            .declareUFImpl(
+                "__unnamed_arr", toSolverType(pElementType), List.of(toSolverType(pIndexType)));
```

Comment:
Is this extra UF really necessary?
How can we create two arrays with default values, when they have distinct types? (test missing)
Is this name whitelisted as normal variable name and excluded in model generation? (test missing)

Reply:
I don't see another way of creating a non-initialized array literal. I added some extra tests to verify that these do not create problems with e.g. vars with the same name. It turns out that only Z3 works well with such literals (without a default value), and only Z3, MathSAT and CVC5 work with initialized literals.